src/core/server/elasticsearch/legacy/elasticsearch_client_config.ts

```diff
@@ -180,7 +179,9 @@ function getDurationAsMs(duration: number | Duration) {
   return duration.asMilliseconds();
 }
 
-function getLoggerClass(log: Logger, logQueries = false) {
+function getLoggerClass(log: Logger) {
+  const queryLogger = log.get('query');
+
```

Comment:
> Moves ES query and error logs to a new logging context: elasticsearch.query

AFAIK, the client's context is prefixed by the client's type/name

https://github.com/elastic/kibana/blob/170a2956c84d46262a3ae310f89aee6f2580860a/src/core/server/elasticsearch/elasticsearch_service.ts#L126-L132

Meaning that the context for the default client is `elasticsearch.data.query`, and that any custom client will have another, specific logging context.

I guess we may need to change that?

Reply:
Thanks for catching that. I do want these query logs to be under a specific context so that other types of logs that we add (eg. errors, audit) can be configured separately. However, I also want it to be easy to disable or enable ALL query logs, regardless of plugin. I will change this to pass in `this.coreContext.logger.get('elasticsearch')` along with the `type` as a separate parameter to `ClusterClient.prototype.constructor` and then construct the sub-loggers individually so that the type-specific part of the context is always last in the context.